transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -482,7 +482,8 @@ public void run() {
         return ctx;
     }
 
-    private static void remove0(AbstractChannelHandlerContext ctx) {
+    private void remove0(AbstractChannelHandlerContext ctx) {
+        assert Thread.holdsLock(this);
```

Comment:
Doesn't this imply that remove0 should be `synchronized`?

Reply:
@nitsanw it imply that `remove0` should only be called from a scope which is `synchronized(this)`

Reply:
Isn't that exactly what a `synchronized` method does?

Reply:
sure but why mark it `synchonized` when it is always called from a `synchronized` scope anyway ? If it is not called from a `synchronized` scope its a bug... Like here ;)

Reply:
I understand this used to be a static method, and therefore you could not mark it as `synchronized`, but now that you can it would AFAICT prevent this bug far less verbosely (it would make the old code correct, make the assertion redundant). Is there a reason not to use `synchronized` here? :-)

Reply:
@nitsanw I am more sure :) In many cases the synchronized need to not only have this code in but much more so I kind of preferred it to not have it on the method only and so make it more "verbose".